core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/userdata/UpdateUserDataHandler.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.cloudbreak.reactor.handler.userdata;
+
+import static com.sequenceiq.cloudbreak.core.flow2.stack.update.userdata.UpdateUserDataEvents.UPDATE_USERDATA_FAILED_EVENT;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.reactor.api.event.stack.userdata.UserDataUpdateFailed;
+import com.sequenceiq.cloudbreak.reactor.api.event.stack.userdata.UserDataUpdateRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.stack.userdata.UserDataUpdateSuccess;
+import com.sequenceiq.cloudbreak.service.image.userdata.UserDataService;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class UpdateUserDataHandler extends ExceptionCatcherEventHandler<UserDataUpdateRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateUserDataHandler.class);
+
+    @Inject
+    private UserDataService userDataService;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(UserDataUpdateRequest.class);
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<UserDataUpdateRequest> event) {
+        LOGGER.error("Updating user data in the stack's image entity has failed", e);
```

Comment:
Strictly speaking, in `core` module it is stored as `component`
It is true however that in `freeipa` it is stored as an image entity.

Reply:
ok